### PR TITLE
[#402] adds ability to refresh a user access token automatically and retry last request for user resources

### PR DESCRIPTION
**Pull request by cristi-badila**

### CODE REVIEW

File: lib/omni_api/config.rb [hunk 1]
```diff
@@ -22,11 +22,14 @@ def self.client_access_token=(token)
   # rubocop:disable Style/ClassVars
   def self.user_access_token=(token)
     @@user_access_token = "bearer #{token}"
-  end
+    end
 
   mattr_accessor :base_url
   self.base_url = ''
 
+  mattr_accessor :user_refresh_token
```
Comment (alexandru-calinoiu) [inline lib/omni_api/config.rb]: Please group this with the `user_token`

File: lib/omni_api/resources/user_authorization_error_handler.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module OmniApi
+  module Resources
+    class UserAuthorizationErrorHandler
+      def handle(error, connection, arguments)
+        raise error unless error.is_a?(ActiveResource::UnauthorizedAccess)
+        token = Oauth2::Token.refresh_for(OmniApi.config.user_refresh_token) rescue nil
+        raise error unless token && token.persisted?
```
Comment (alexandru-calinoiu) [inline lib/omni_api/resources/user_authorization_error_handler.rb]: This looks weird, what will the use case be? 

I prefer the simple `refresh_token!` at least we know the actual error.

**Comment by cristi-badila** [comment]
@alexandru-calinoiu updated pull request

**Comment by cristi-badila** [comment]
merged

**Pull request closed without merging**